### DISCUSSION TITLE
Have the History "Message" show the diff when updating

### DISCUSSION THREAD

**Discussion by RyanTG** [railsadminteam/rails_admin#3618] [Q&A]
Hi, I'm upgrading and moving from the history adapter to paper_trail. Seemingly, the last thing to resolve:

Previously with the history adapter, after editing a record, the "Message" field in history would (fairly) clearly state what was edited. For example: 

![image](https://user-images.githubusercontent.com/632682/230551856-0d91dc15-412c-4dab-a3eb-161e82b53615.png)

But so far with paper_trail, this Message field only states "Update" in it. This appears to be the "Event", as show here https://github.com/railsadminteam/rails_admin/blob/5a958e46e8620fba320e04347d326d058c07d396/lib/rails_admin/extensions/paper_trail/auditing_adapter.rb#L14

I would rather the message show the object_changes when the event is an update.

Example object_changes values from a transaction.

```
          object_changes           
-----------------------------------
 ---                              +
 updated_at:                      +
 - 2023-04-07 00:23:29.000000000 Z+
 - 2023-04-07 05:50:26.110112386 Z+
 operator_id:                     +
 - 237                            +
 - 435                            +

```

Thanks!

**Comment by RyanTG** [comment]
To tack onto this: I just realized that paper_trail is tracking all edits to the models that have has_paper_trail - this isn’t what I want. I only care to track admin edits made within rails admin, in the history tabs. In rails_admin v2 using the history adapter, this was perfect! We had a great record of activity made within rails_admin. paper_trail seems like a step backward to me.

**Comment by RyanTG** [comment]
I opened the class in an initializer and changed what it shows in the message when the event is an update. But this is still a big step back from the old rails_admin v2 history.

```
RailsAdmin::Extensions::PaperTrail::VersionProxy.module_eval do
  def message
    if @version.event == "update"
        @message = @version.object_changes
    else
        @message = @version.event
    end
    @version.respond_to?(:changeset) && @version.changeset.present? ? @message + ' [' + @version.changeset.to_a.collect { |c| "#{c[0]} = #{c[1][1]}" }.join(', ') + ']' : @message
  end
end
```

Still not sure how to fix my other issue of only wanting to show history from administrators.

**Comment by brymacto** [comment]
I think this is caused by https://github.com/paper-trail-gem/paper_trail/pull/1399.  

You could verify by downgrading paper trail.  But when loading `changeset` from the paper trail version, it's now using yaml safe loading.  fix would be to configure rails to [allow values in yaml deserialization](https://github.com/paper-trail-gem/paper_trail/pull/1397).
